Add m3dt Template Reference feature test

kashifest opened the pull request:
Add m3dt Template Reference feature test

kashifest commented:
/retest

kashifest commented:
/test markdownlint

kashifest commented:
/test-features

fmuyassarov commented:
/test markdownlint

russellb commented:
/test markdownlint

kashifest commented:
/test-features

kashifest commented:
/test markdownlint

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

maelk commented on vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:
```diff
@@ -220,6 +220,43 @@
+    register: m3d_name
+  
+  - name: Check if Metal3Data refers to the old template
+    shell: "kubectl get m3d -n {{ NAMESPACE }}  {{ m3d_name }} -ojson | jq '.spec.templateReference=={{ CLUSTER_NAME }}-workers-template'"
```
what is jq returning ? will the whole command fail if it is not true ? or should there be an ansible exit condition afterwards ?

kashifest replied:
bool. I am not sure, I wanted to check it in CI. But is that check enough or do I need to check something else

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-integration
/test-centos-integration

kashifest commented:
/test-features
/test-integration
/test-centos-integration

kashifest commented:
/test-features
/test-integration
/test-centos-integration

kashifest commented:
/test-features

furkatgofurov7 commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-features

kashifest commented:
/test-integration
/test-centos-integration

kashifest commented:
/test-features

kashifest commented:
/test-features

furkatgofurov7 commented on vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:
```diff
@@ -220,6 +220,50 @@
+
+  - name: Check if one Metal3Data refers to the old template
+    shell: kubectl get m3d -n metal3 -ojson | 
+              jq -r '[ .items[] | select (.spec.templateReference=="test1-workers-template")|
```
```suggestion
              jq -r '[ .items[] | select (.spec.templateReference=="{{ CLUSTER_NAME }}-workers-template")|
```

kashifest replied:
Thanks, since it is a cosmetic change. I ll do it in some future PR since passing the feature test now is a big issue.

kashifest replied:
Done

furkatgofurov7 reviewed (commented):
Looks good to me, not blocking nit:

smoshiur1237 reviewed (approved):
It looks good to me.

smoshiur1237 commented:
/lgtm

Xenwar commented on vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:
```diff
@@ -220,6 +220,50 @@
+              jq '.|del(.status)' | 
+              kubectl apply -f-
+
+  - name: Edit MachineDeployment to point to the new m3mt
```
This is a good descriptive.
The other task names can also be improved like it as the current names are referring to the operations, not purpose.

kashifest replied:
Thanks done

Xenwar commented on vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:
```diff
@@ -220,6 +220,50 @@
     register: ready_hosts
     until: ready_hosts.stdout == "1"
+    
+    #### m3dt template reference check
```
Remove or change this comment to a more descriptive one

kashifest replied:
Removed. Thanks

kashifest commented:
/test-integration
/test-centos-integration
Thanks for the review folks. I have addressed the comments. PTAL look again @furkatgofurov7 @Xenwar @fmuyassarov @smoshiur1237 @maelk

furkatgofurov7 commented on vm-setup/roles/v1aX_integration_test/tasks/remediation.yml:
```diff
@@ -220,6 +220,49 @@
     until: ready_hosts.stdout == "1"
+    
+    ## Start Metal3DataTemplate reference test
+  - name: Get the Metal3DataTemplate, Edit name , Add templateReference and apply it 
```
```suggestion
  - name: Get the Metal3DataTemplate, Edit name, Add templateReference and apply it 
```

furkatgofurov7 reviewed (commented):
/lgtm

kashifest commented:
/test-features
